server.js

```diff
@@ -1633,7 +1633,100 @@ app.get('/adminAnalytics', function(req,res) {
     });
 });
 
+app.post('/submitFeedback', function(req, res){
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type === common.userTypes.ADMIN) {
+        return res.status(403).send(common.getError(1002));
+    }
+
+    logger.log(common.formatString('Feedback from {0} regarding {1}', [req.session.user._id, req.body.subject]));
+
+    users.addFeedback(req.session.user._id, req.body.subject, req.body.message, function(err, result) {
+        if (err) {
+            logger.error(err);
+            return res.status(500).send(common.getError(8000));
+        }
+
+        return res.status(201).send('User feedback submitted');
+    });
+});
+
+app.get('/feedback', function(req, res){
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type !== common.userTypes.ADMIN) {
+        return res.status(403).send(common.getError(1002));
+    }
+
+    logger.log(common.formatString('Getting feedback for {0}', [req.session.user.username]));
+
+    users.getFeedback(function(err, result) {
+        if (err) {
+            logger.error(err);
+            return res.status(500).render('feedback-view', {
```

Comment:
not sure what is happening here, please explain

Reply:
was being breakfast, mb